datafusion/src/scalar.rs

```diff
@@ -453,6 +477,26 @@ macro_rules! eq_array_primitive {
 }
 
 impl ScalarValue {
+    /// Create a decimal Scalar from value/precision and scale.
+    pub fn try_new_decimal128(
+        value: i128,
+        precision: usize,
+        scale: usize,
+    ) -> Result<Self> {
+        // make sure the precision and scale is valid
+        // TODO const the max precision and min scale
```

Comment:
The arrow spec doesn't seem to say anything about min/max valid scales: https://github.com/apache/arrow/blob/master/format/Schema.fbs#L181-L190

I poked a little around in arrow-rs and I also didn't find any limits on the precision or scale either

Reply:
If we use 128bit to represent a decimal value, the max `precision` is 38.
The `scale` is always greater or equal to 0 and less than or equal to `precision`.
In the arrow-rs, the decimal is represented by `i128` with `precision` and `scale`.
So we can add the `MAX_PRECISION = 38` in the datafusion.

Reply:
Just to make sure I got this correctly.
The spec doesn't say anything about the the precision/scale of `DECIMAL`, but we're implementing `DECIMAL128` here so 38 should be the max precision.
After https://github.com/apache/arrow-rs/issues/131 is implemented, we can implement `DECIMAL256` type in datafusion and the max precision will be 76.

Reply:
> 

Yes, now in the arrow-rs the decimal256 has not been implemented.
In the arrow-go, it is also not implemented too.
Just in the arrow-java and arrow-c++, the bitwith of 256 has been implemented.
@capkurmagati

Reply:
we can make this const in other pull request